src/qt3utils/datagenerators/plescanner.py

```diff
@@ -0,0 +1,244 @@
+import tkinter as tk
+import numpy as np
+import time
+import logging
+from qt3utils.applications.controllers.nidaqedgecounter import QT3ScopeNIDAQEdgeCounterController
+from qt3utils.nidaq.customcontrollers import VControl
+from qt3utils.applications.controllers.wavemeter_controller import WavemeterController
+
+logger = logging.getLogger(__name__)
+
+
+class PleScanner:
+    """
+    Base class with function methods which any PLE experiment will need to operate
+    Override this class for specific types of PLE experiments
+    """
+
+    def __init__(self, wavelength_controller):
+        self.running = False
+        self.current_t = 0
+        self.vmin = float(wavelength_controller.minimum_allowed_position)
+        self.vmax = float(wavelength_controller.maximum_allowed_position)
+        self.tmax = 100
```

Comment:
It is specified in the GUI, not the YAML, so the value here doesn't actually matter as it is set in the main.py from the user. I replaced them with None here so as to avoid confusion in the future.